source/common/ssl/context_config_impl.cc

```diff
@@ -44,5 +44,10 @@ ContextConfigImpl::ContextConfigImpl(const Json::Object& config) {
   server_name_indication_ = config.getString("sni", "");
 }
 
+ServerContextConfigImpl::ServerContextConfigImpl(const Json::Object& config)
+    : ContextConfigImpl(config) {
+  require_client_certificate_ = config.getBoolean("require_client_certificate", false);
```

Comment:
I'd rather avoid add unrelated changes (to `ContextConfigImpl`) here. We can handle this later in a separate commit.